pkg/metrics/platform_test.go

```diff
@@ -150,16 +150,28 @@ var _ = Describe("PlatformMetrics", func() {
 
 })
 
-func getGaugeValue(platform, metricName string) (int, error) {
+func getGaugeValue(platform, metricName, namespace string) (int, error) {
 	mfs, err := metrics.Registry.Gather()
 	if err != nil {
 		return 0, err
 	}
 	for _, mf := range mfs {
 		if mf.GetName() == metricName {
 			for _, m := range mf.GetMetric() {
-				if m.Gauge != nil && m.Label[0].GetValue() == platform {
-					return int(m.Gauge.GetValue()), nil
+				if m.Gauge != nil {
+					hasPlatform := false
+					hasNamespace := false
+					for _, l := range m.Label {
+						if l.GetName() == "platform" && l.GetValue() == platform {
+							hasPlatform = true
+						}
+						if l.GetName() == "namespace" && l.GetValue() == namespace {
+							hasNamespace = true
+						}
+					}
+					if hasPlatform && hasNamespace {
+						return int(m.Gauge.GetValue()), nil
+					}
```

Comment:
minor: idea to simplify this block:

```suggestion
				if m.Gauge == nil {
					continue
				}

				if hasLabel(m.Label, "platform", platform) && hasLabel(m.Label, "namespace", namespace) {
					return int(m.Gauge.GetValue()), nil
				}
```

where `hasLabel` is as follow:

```golang
import io_prometheus_client "github.com/prometheus/client_model/go"

func hasLabel(lp []*io_prometheus_client.LabelPair, name, value string) bool {
	return slices.ContainsFunc(lp, func(l *io_prometheus_client.LabelPair) bool {
		return l.GetName() == name && l.GetValue() == value
	})
}
```